### ISSUE TITLE
Agent ServicePodWorker gets confused by service restarts

### ISSUE THREAD

**Issue by SpComb** [kontena/kontena#2781]
The `ServicePodWorker` does not know anything about the [`ContainerHealthCheckWorker#restart_container`](https://github.com/kontena/kontena/blob/3891ac5ee1d680213ae895e67123345298ad40b1/agent/lib/kontena/workers/container_health_check_worker.rb#L110) / [RPC `/service_pods/restart`](https://github.com/kontena/kontena/blob/45cf5ffaa02493ff0ee180fbcae33379d5f41c5b/agent/lib/kontena/rpc/service_pods_api.rb#L11) -> `ServicePods::Restarter` calls. It sees the container `die` event and thinks that the container has exited and needs to be restarted.

That logs `Kontena::Workers::ServicePodWorker: restarting sleep-test-1 because it has stopped` and triggers a no-op `apply`, where it just determines that the container is actually already running again. Then it sees the `stop`, `start` and `restart` events.

```
D, [2017-09-07T12:59:51.961906 #1] DEBUG -- Kontena::RpcServer: rpc request: Kontena::Rpc::ServicePodsApi#restart ["59b14290abe74500089e03e6", 1]
I, [2017-09-07T12:59:51.982587 #1]  INFO -- Kontena::ServicePods::Restarter: restarting service: /sleep-test-1
D, [2017-09-07T12:59:52.005168 #1] DEBUG -- Kontena::Workers::ServicePodWorker: container event: kill
D, [2017-09-07T13:00:02.008805 #1] DEBUG -- Kontena::Workers::ServicePodWorker: container event: kill
D, [2017-09-07T13:00:02.041565 #1] DEBUG -- Kontena::Workers::ServicePodWorker: container event: die
I, [2017-09-07T13:00:02.042057 #1]  INFO -- Kontena::Workers::ServicePodWorker: restarting sleep-test-1 because it has stopped
D, [2017-09-07T13:00:02.042428 #1] DEBUG -- Kontena::Workers::ServicePodWorker: restart triggered (from 2017-09-07 13:00:02 UTC)
D, [2017-09-07T13:00:02.042806 #1] DEBUG -- Kontena::Workers::ServicePodWorker: state of sleep-test-1: running
D, [2017-09-07T13:00:03.276990 #1] DEBUG -- Kontena::Workers::ServicePodWorker: state is in-sync: running
D, [2017-09-07T13:00:03.380802 #1] DEBUG -- Kontena::Workers::ServicePodWorker: container event: stop
D, [2017-09-07T13:00:03.399293 #1] DEBUG -- Kontena::Workers::ServicePodWorker: container event: start
D, [2017-09-07T13:00:03.399550 #1] DEBUG -- Kontena::Workers::ServicePodWorker: container event: restart
```

**Comment by SpComb** [comment]
Bumping from 1.4.0 -> 1.4.1 because at this point I'm not aware of any actual user-visible side-effects of this confusion. The container still gets restarted, and the pod worker doesn't try to re-restart it.

**Comment by SpComb** [comment]
Part of a much broader issue about multiple different actors in the agent poking the container with restart operations, not happening for 1.4.x

PR to fix this for restart operations is #2780, but that still has issues with queuing up of healthcheck restarts (healthcheck fails -> start restarting container -> healthcheck fails -> queue restart -> container restarts complete -> restart it again because healthcheck failed)
